### PR TITLE
add iotaSlice

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

**Comment by 9il** [comment]
ping @wilzbach, this is fix for #22

File: source/mir/ndslice/selection.d [hunk 1]
```diff
@@ -1556,18 +1557,28 @@ IndexSlice!N indexSlice(size_t N)(auto ref size_t[N] lengths)
 }
 
 ///
-@safe @nogc pure nothrow unittest
+@safe pure nothrow unittest
```
Comment (wilzbach) [inline source/mir/ndslice/selection.d]: I start to dislike  the Allocator now, but it might be worth to think about adding support for it?

Comment (9il) [inline source/mir/ndslice/selection.d]: What do you mean? `makeSlice` and `dispose` was added to `ndslice.slice`. `indexSlice` is lazy and `@nogc`.

Comment (wilzbach) [inline source/mir/ndslice/selection.d]: > indexSlice is lazy and @nogc.

You removed the check for nogc here.

Comment (9il) [inline source/mir/ndslice/selection.d]: Yes, because `[[[0, 0], [0, 1], [0, 2]],  [[1, 0], [1, 1], [1, 2]]]` allocates

Comment (wilzbach) [inline source/mir/ndslice/selection.d]: `static immutable`?

Comment (9il) [inline source/mir/ndslice/selection.d]: :-( ok

File: source/mir/ndslice/selection.d [hunk 2]
```diff
@@ -1534,6 +1534,7 @@ pure nothrow unittest
 
 /++
 Returns a slice, the elements of which are equal to the initial index value.
+This is multidimensional analog of $(LINK2 std_range.html#iota, std.range.iota).
```
Comment (wilzbach) [inline source/mir/ndslice/selection.d:1537]: Maybe you could add a see also to iota slice? 
Moreover I think that if you at the multidimensional analog of iota, one might think about iotaSlice, so maybe pointing out that the index of reseted for every dimension could help to solve the confusion? 
And then "for a continuous index, see iotaSlice"

**Comment by wilzbach** [comment]
> ping @wilzbach, this is fix for #22

Looks great - exactly what #22 was about :)

**Comment by 9il** [comment]
Thank you for the review!

**Pull request merged**